apps/server/src/services/rundown-service/rundownUtils.ts

```diff
@@ -63,9 +63,16 @@ export function getEventWithId(eventId: string): OntimeEvent | undefined {
  * @param {string} cue
  * @return {object | undefined}
  */
-export function getEventWithCue(cue: string): OntimeEvent | undefined {
-  const timedEvents = getTimedEvents();
-  return timedEvents.find((event) => event.cue.toLowerCase() === cue.toLowerCase());
+export function getNextEventWithCue(cue: string, currentEventId?: string): OntimeEvent | undefined {
+  // const loaded = curr
+  const timedEvents = getPlayableEvents();
+  const comparCue = cue.toLowerCase();
+  if (currentEventId) {
+    const currentIndex = timedEvents.findIndex((event) => event.id === currentEventId);
+    const futureEvents = timedEvents.slice(currentIndex);
+    return futureEvents.find((event) => event.cue.toLowerCase() === comparCue);
+  }
+  return timedEvents.find((event) => event.cue.toLowerCase() === comparCue);
 }
```

Comment:
this only seams to be used inside RundownService, I would suggest that the consumer passes in a cue and a current index. the function would be much simpler just looking for a match after the index. 

```ts
export function getNextEventWithCue(targetCue: string, currentEventIndex = 0): OntimeEvent | undefined {
  const timedEvents = getPlayableEvents();
  const lowerCaseCue = targetCue.toLowerCase();

  for (let i = currentEventIndex; i < timedEvents.length; i++) {
    const event = timedEvents.at(i);
    if (event && event.cue.toLowerCase() === lowerCaseCue) {
      return event;
    }
  }
}
```


I think the function would be even better if it got passed the timedEvents, as the ones in the other rundownUtils file does.
Not sure why we are keeping two definitions of the same functions. the ones where everything is passed are much easier to test